src/main/scala/org/apache/mesos/chronos/scheduler/jobs/stats/JobStats.scala

```diff
@@ -360,7 +353,16 @@ class JobStats @Inject()(clusterBuilder: Option[Cluster.Builder], config: Cassan
     }
   }
 
-  def jobQueued(job: BaseJob, attempt: Int) {
+  override def onEvent(event: JobEvent): Unit = event match {
+    case JobRemoved(job) => removeJobState(job)
+    case JobQueued(job, taskId, attempt) => jobQueued(job, taskId, attempt)
+    case JobStarted(job, taskStatus, attempt) => jobStarted(job, taskStatus, attempt)
+    case JobFinished(job, taskStatus, attempt) => jobFinished(job, taskStatus, attempt)
+    case JobFailed(job, taskStatus, attempt) => jobFailed(job, taskStatus, attempt)
+    case _ => // do nothing
```

Comment:
PartialFunctions sounds good or at least there should be a log info message letting us know that the event is not currently handled.